packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh

```diff
@@ -105,8 +105,8 @@ regen=
 #################################
 # Tests to symbolically execute #
 #################################
-# Missing: OptimismPortalKontrol.prove_proveWithdrawalTransaction_paused
-test_list=( "OptimismPortalKontrol.prove_finalizeWithdrawalTransaction_paused" \
+test_list=( "OptimismPortalKontrol.prove_proveWithdrawalTransaction_paused" \
+            "OptimismPortalKontrol.prove_finalizeWithdrawalTransaction_paused" \
```

Comment:
Since we now run all tests by default, can we remove the need for the `--match-test` flag all together, and update this so any extra args passed to the script are forwarded to `kontrol prove`? This wasy I can do `./run-kontrol.sh --match-test prove_proveWithdrawalTransaction_paused` to run a single test

Reply:
Sure thing! I would be more inclined to keep it as it is, but if you prefer feeding the tests to `run-kontrol.sh` directly it certainly can be done! Also, do you want to add `--match-test` for each test, or just the name of the tests?
Also also, should the behavior when no parameters are passed (`./run-kontrol.sh dev`) be to execute all the tests, or throw an error? For CI purposes it would seem best to do the former

Reply:
My thinking was that the best UX for `./run-kontrol.sh` is to match the UX of `forge test` as much as possible, so by default we'd run all detected tests, and pass flags to filter. So the assumption I made in my comment above is that by default `kontrol prove` would find all `prove_*` tests and run the, but maybe that's incorrect?

Reply:
That seems reasonable to me! But I would have the following 3 behaviors instead of 2:
- `./run-kontrol.sh [container/local/dev] name_of_test`: execute `name_of_test` in the mode specified
- `./run-kontrol.sh [container/local/dev]`: execute all tests specified in `run.kontrol.sh`
- `./run-kontrol.sh [container/local/dev] all`: execute `kontrol prove`, matching all `prove_*`, `test_*` and `check_*` functions

Does this seem alright? The main point is to have some control (no pun intended lol) at the script level of what proofs are being executed and not rely entirely on passing arguments to the script. This can end up in a particularly long command if we're feeding it around 15 proofs

Also, it turns out that for the moment, `kontrol prove` without any `--match-test` will only match `test_*` functions, but we're already addressing this, so it will land soon in Kontrol

Reply:
The only reason I'd like to have some control at the script level still is when developing proofs because we might want to exclude/include some, which may lead to an arbitrary subset of the proofs being executed. I think it would be a bit cumbersome to edit a large command when doing proof development or testing out stuff rather than just commenting some lines in `run-kontrol.sh`.
I don't see anything wrong with what you're suggesting! So, maybe we could have the two cases you're describing plus
- `./run-kontrol.sh [container/local/dev] script`: execute all tests specified in `run-kontrol.sh` (it doesn't need to be specifically `script`)

What do you think? If you would rather not have this option I'm describing, I'm okay with leaving it out

Reply:
Ultimately, the `run-kontrol.sh` script must fit you, so if you find it more intuitive/consistent to work with `forge` UX, that's fine with me! I do think it's a good idea! It's just that, from what I'm used to do in proof development, I see keeping the `script` option useful.
A case for having the script option is, for instance, to execute all proofs that only have a `bytes` argument and no `bytes[]`. In that case, we're only excluding one proof, which we can also do by adding `vm.skip()` to it, but I think I'd lean more toward not modifying the code if we could just modify the execution parameters

Reply:
Makes sense, let's keep the `script` option then. So I think the final state we have is:
- no args: run all tests
- script arg: run the ones specified in the script
- otherwise, forward all matching args like `--match-test`